### PR TITLE
Drop python 2 and switch to automatic version numbers

### PR DESCRIPTION
**Pull request by djhoese**
This PR does a few important things:

- Drop explicit Python 2 support by changing the `python_requires` kwarg in `setup.py`.
- Switch from hardcoded version numbers to `setuptools_scm`
- Update tests to work with newer versions of ci-helpers and test on Python 3.8
- Fix README.rst not being read properly in the setup.py or included in the sdist tarball
- Add release instructions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,28 +12,20 @@ env:
...
     os: linux
     language: generic
-  - env: PYTHON_VERSION=2.7
-    os: osx
-    language: generic
-  - env: PYTHON_VERSION=3.6
-    os: linux
-    language: generic
-  - env: PYTHON_VERSION=3.6
+  - env: PYTHON_VERSION=3.8
```
Comment (mraspaud) [inline .travis.yml]: Should we also have 3.7? I think it's good to have two versions at least, it's easier to see if a failure is just a glitch.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Setuptools SCM version.py file
+pydecorate/version.py
```
Comment (mraspaud) [inline .gitignore]: Wasn't the file deleted in this PR ?

**Comment by djhoese** [comment]
I should have mentioned that this implementation differs from what we see in Satpy where we *don't* create a `version.py` file and instead load it dynamically on import. This is not preferred as this dynamic version stuff does not scale and has performance issues on large environments.

File: .gitignore (hunk 2 above)
Comment (djhoese) [reply .gitignore]: It is auto-generated by setuptools_scm when you install the package.

File: .travis.yml (hunk 1 above)
Comment (djhoese) [reply .travis.yml]: I don't think it is necessary, but I added it just for you :smile:

File: .gitignore (hunk 2 above)
Comment (mraspaud) [reply .gitignore]: ok, I understand then.

**Comment by djhoese** [comment]
@mraspaud I reran the failing test and now everything is passing. If you are ok merging this then we could make a new release.

**Pull request merged**